plugins/carlabase/carla.cpp

```diff
@@ -449,10 +443,15 @@ bool CarlaInstrument::handleMidiEvent(const MidiEvent& event, const MidiTime&, f
 
 PluginView* CarlaInstrument::instantiateView(QWidget* parent)
 {
+// Disable plugin focus per https://bugreports.qt.io/browse/QTBUG-30181
+#ifdef CARLA_OS_MAC
```

Comment:
there is no need to duplicate code here, we can have this instead:
```
#ifndef CARLA_OS_MAC
    if (QWidget* const window = parent->window())
        fHost.uiParentId = window->winId();
    else
#endif
        fHost.uiParentId = 0;
```

Reply:
This was a style preference, not an oversight.  Precompiler checks don't add additional logic to runtime, it's only a single redundant line, which is subject to change if the focus feature is needed and if we need to use fancy Cocoa calls.

I'm fine consolidating this to match other areas of our code, but this was done for readability.  @PhysSong your call.

Reply:
I prefer what @falkTX suggested, but I'm fine with any of the two choices.

Reply:
I prefer what @falkTX suggested, but I'm fine with any of the two choices.

Reply:
Done via 0838c9b.